### PR TITLE
AppBar: Add support for custom branding

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: https://app.shortcut.com/cartoteam/story/344453/whitelabel-issue-icons-and-other-elements-are-not-respecting-the-app-bar-text-color
[sc-344453]

Add basic support for colors customization in AppBar core components.

Screenshot 2023-10-19 at 11 55 55

### CODE REVIEW

File: packages/react-ui/storybook/stories/molecules/Table.stories.js [hunk 1]
```diff
@@ -52,7 +54,15 @@ const rows = [
 const options = {
   title: 'Molecules/Table',
   component: Table,
-  argTypes: {},
+  argTypes: {
```
Comment (vmilan) [inline packages/react-ui/storybook/stories/molecules/Table.stories.js:57]: Offtopic... story improved as a design request

**Review by mdeividas — APPROVED** [review]
LGTM

**Review by zbigg — APPROVED** [review]

File: packages/react-ui/src/components/organisms/AppBar/BurguerMenu.js [hunk 2]
```diff
@@ -12,22 +12,32 @@ const Menu = styled('div')(({ theme }) => ({
...
-export default function BurguerMenu({ onClickMenu }) {
+const MenuDivider = styled(Divider, {
+  shouldForwardProp: (prop) => prop !== 'color'
+})(({ color, theme }) => ({
+  ...(color && {
+    borderColor: alpha(color, 0.12)
+  })
+}));
+
+export default function BurguerMenu({ onClickMenu, iconColor }) {
```
[the remark this answers is not in the document]
Comment (vmilan) [reply packages/react-ui/src/components/organisms/AppBar/BurguerMenu.js]: Nice catch, fixed, thanks!

File: packages/react-ui/src/components/organisms/AppBar/AppBar.d.ts [hunk 3]
```diff
@@ -9,6 +9,8 @@ export type AppBarTypeMap<D extends React.ElementType<any> = 'header'> = MuiAppB
     secondaryText?: React.ReactNode;
     onClickMenu?: (event: React.MouseEvent) => void;
     showBurgerMenu?: boolean;
+    backgroundColor?: string;
```
Comment (VictorVelarde) [inline packages/react-ui/src/components/organisms/AppBar/AppBar.d.ts:12]: Oks, I see the gap, 👍🏻

**Review by VictorVelarde — APPROVED** [review]
👍🏻 Nice work

**Pull request merged**